[sarang_daddy] 2주차

sarangdaddy opened the pull request:
2주차 학습계획
- [x] 월요일
  - [x] 1주차 리뷰 내용 적용
  - [x] 언론사 리스트 캐러셀 마무리
 
- [x] 화요일
  - [x] 현황 파악하기
  - [x] 상태관리 고민하기
 
<현황>
<img width="600px" alt="image" src="https://user-images.githubusercontent.com/109648042/231120940-8bf59bca-ff61-4ff5-85a5-cd56383fa58e.png">
 
- [ ] 수요일
  - [x] MVC, FLUX 학습
  - [ ] 상태관리 가능한 모델 설계
 
- [ ] 목요일  
  - [ ] 구독, 취소 기능 고민....

sarangdaddy committed:
[Docs] template literal사용, 주석 제거

sarangdaddy committed:
[Feat] journalCarousel 파일 생성

sarangdaddy committed:
[Rename] getData API 파일 하나로 통합

- headLine, journal 데이터를 받아오던 api js 파일을 하나로 통일

sarangdaddy committed:
[Refactor] 1주차 리뷰내용 적용

- 모든 로직 함수나 클래스로 옮기기
- 데이터 받아오는 파일(api) 하나로 통합하기
- 주석, 필요없는 코드 등 삭제

sarangdaddy committed:
[Feat] 언론사 캐러셀 기능 추가

- 왼쪽 버튼 클릭시 왼쪽 페이지로 이동
- 오른쪽 버튼 클릭시 오른쪽 페이지로 이동
- 왼쪽 끝 페이지로 오면 왼쪽 버튼 사라짐
- 오른쪽 끝 페이지로 오면 오른쪽 버튼 사라짐

sarangdaddy committed:
[Refactor] journalCarousel 코드 수정

- 매직넘버 제거
- 페이지 프로퍼티 추가

sarangdaddy committed:
[Refactor] main로직 함수안으로 가져오기

- 변수명 수정

realsnoopso commented on src/api/getData.js:
```diff
@@ -9,3 +10,13 @@ export const getHeadLine = (URL) => {
+      return journal.makeJournal();
+    })
+    .catch((error) => console.error(`fetch 에러! ${error}`));
+};
```
.then 말고 await로 바꾸면 더 보기 좋을 것 같아요!

realsnoopso commented on src/components/journalList/journalListView.js:
```diff
@@ -1,13 +1,15 @@
 
-const journalURL = " http://localhost:3000/journal";
+const createNewsStandJournal = () => {
+  const journalURL = " http://localhost:3000/journal";
```
띄어쓰기가 있어요!

realsnoopso commented on src/api/getData.js:
```diff
@@ -9,3 +10,13 @@ export const getHeadLine = (URL) => {
+  return fetch(URL)
+    .then((response) => response.json())
+    .then((jsonData) => {
+      const journal = new Journal(jsonData);
+      return journal.makeJournal();
```
api 폴더 안에 있는 파일인데 Journal 이라는 html 태그를 생성하는 클래스가 여기에서 선언되는게 어색한거 같아요.
api 파일이라고 하면 로직만 담고 있을 것 같거든요. view와 controller 영역을 분리해보면 좋을 것 같아요.

realsnoopso commented on src/getData/journalData.js:
```diff
@@ -1,19 +1,23 @@
+    const obj = {
+      journalId: id,
+      journalAlt: listEl.alt,
+      journalSrc: listEl.currentSrc,
+    };
 
-  result.push(obj);
-  id++;
-});
+    result.push(obj);
```
forEach 대신 map을 쓰면 .push 메소드를 사용하지 않고도 array 를 반환할 수 있을 것 같아요.

realsnoopso commented on src/main.js:
```diff
@@ -1,11 +1,24 @@
+
+  // main에 헤더. 헤드라인, 언론사 추가
+  mainSpace.append(
+    createNewsStandHeader(),
+    createNewsStandHeadLine(),
```
마찬가지로 view에 관한 처리가 여기 오니까 어색하다고 느껴져요. 아니면 여기가 controller 역할을 담당하는 파일인 걸까요? 
+ getData라는 폴더명이 api/components 들과 위계가 안맞는 것처럼 느껴져서 (getData만 동사이고 나머지는 명사임) 폴더명을 수정하는 것도 고려해보면 좋을 것 같아요

realsnoopso commented on src/components/journalList/journalCarousel.js:
```diff
@@ -0,0 +1,61 @@
+  const currentPosition = this.page * -WIDTH_PER_PAGE;
+  this.container.style.transform = `translateX(${currentPosition}px)`;
+
+  if (this.page === FIRST_PAGE) {
+    this.leftBtn.style.display = "none";
+  } else if (this.page === LAST_PAGE) {
+    this.rightBtn.style.display = "none";
+  } else {
+    this.leftBtn.style.display = "block";
+    this.rightBtn.style.display = "block";
+  }
```
if (this.page === FIRST_PAGE) {
    ...
    return ...
}

if (this.page === LAST_PAGE) {
    ...
    return ...
}

... (나머지 코드)

로 else if 와 else 를 수정할 수 있을 것 같아요

realsnoopso commented on src/components/journalList/journalCarousel.js:
```diff
@@ -0,0 +1,61 @@
+    ".journal-carousel-btn:nth-child(2)"
+  );
+
+  this.leftBtn.style.display = "none";
```
이렇게 일일히 style을 조작하는 것보다는 class를 만들고 class 안에 해당 속성을 담은 후 클래스를 조작하는 것이 더 깔끔할 것 같아요. (추후 확장성 고려 차원)

sarangdaddy committed:
[Refactor] 페어리뷰 내용 적용

- api에 있던 생성자 view파일로 이동

sarangdaddy committed:
[Docs] 언론사 상태관리를 위한 기본 틀 구상중

- action : 뷰에서 발생한 상태 변경을 요청한다.
- dispatcher : 액셜을 스토어에게 전달한다.
- store : 언론사의 상태 저장소, dispatcher의 액션으로 상태가 변경되면 뷰에게 알린다.
- view : 상태가 변경되면 리렌더링

crongro reviewed (changes requested):
screen 아래 css 파일들이 보이는데 screen 이라는 이름은 잘 안쓸거 같긴해요.(중요한건 아님)

css파일에 공통으로 사용할만한 유용한 클래스가 부족해보이네요. 중복되는 것을 클래스로 만들어서 재사용해보세요.

async 함수를 활용해서 익혀보시고, try / catch 도 해보세요.

Dispatcher/journalDispatcher.js 는 데이터 요청에 책임이 있는것이지 dispatch 하는 책임은 아닌것 같아요. 따라서 디렉토리/파일이름에 dispatcher 가 붙어야 할까? 고민스럽네요. 

안쓰는 코드 주석은 모두 삭제하고 커밋하세요.

headerlineclass 처럼 구조를 이름에 넣지 마세요.

ES Class 메서드에서는 function 키워드를 생략할 수 있어요.

makeHeadline 보다는 getHeadLineHTML 이 나아보입니다.

Caroulse prototype 패턴은 잘 하셨는데 생성자에서 몇 개 파라미터를 받으면 더 좋을듯.

위 내용 수정해서 다시 커밋/푸시 해서 보내주세요.

sarangdaddy committed:
[Refactor] 언론사 렌더링 함수 분리

- 언론사 영역 전체 렌더링에 포함되어 있더 언론사 리스트 렌더링 분리

sarangdaddy committed:
[Refactor] JournalListView 구조 수정, testFolder 생성

- Journal 영역의 헤더, 캐러설, 렌더 함수 분리
- pattern을 다시 고민해야함. 기록된 코드 저장용 testFolder에 저장

sarangdaddy committed:
[Request] CSS 폴더명 수정

sarangdaddy committed:
[Request] CSS 공통 클래스 생성

- 중복 코드 제거
- 클래스를 이용한 display처리로 수정

sarangdaddy committed:
[Request] async, await 적용

- async의 필요성을 느낄때까지 promise와 then만을 사용해보고 싶어요..

sarangdaddy committed:
[Request] Caroulse 생성자 파라미터로 적용할 엘리먼트 전달

sarangdaddy committed:
[Request] 함수명 수정

- makeHeadLine -> getHeadlineHTML
- makeJournal -> getJournalItem

crongro commented on src/components/journalList/journalClass.js:
```diff
@@ -3,13 +3,38 @@ export class Journal {
-      acc.push(journalDiv);
-      return acc;
-    }, []);
+  getJournalItem() {
```
다음엔 이정도 함수는 더 작은 함수 몇개로 나눠보세요.

crongro reviewed (approved):
수고하셨어요. 머지합니다~
